### DISCUSSION TITLE
New to MSBuild and CodeQL for my C++ Project. (build tools v143 cannot be found)

### DISCUSSION THREAD

**Discussion by turbosmurfen** [github/codeql#7162] [Q&A]
I'm new to CodeQL and MSBuild. I copied the code below for codeql-analysis. And now it won't run.

Somehow I get this error every time CodeQL runs:
`C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(439,5): error MSB8020: The build tools for v143 (Platform Toolset = 'v143') cannot be found. To build using the v143 build tools, please install v143 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [D:\a\spoton\spoton\spoton\spoton.vcxproj]
     2>Done Building Project "D:\a\spoton\spoton\spoton\spoton.vcxproj" (default targets) -- FAILED.
     1>Done Building Project "D:\a\spoton\spoton\spoton.sln" (default targets) -- FAILED.`

And here is the file I use. If someone can tell me what to change: https://github.com/turbosmurfen/spoton/blob/main/.github/workflows/codeql-analysis.yml

Would be really happy if someone could help me with this.

**Comment by hvitved** [comment]
Hi

I don't think this is a CodeQL issue, but rather an issue with building your code using GH actions. Have you tried adding a simple workflow for just building your code?

From the error message it appears that passing in `/p:PlatformToolset="v143"` in your build command does not work, because the `windows-latest` machines does not have a matching MSBuild version installed; what happens if you replace it with `/p:PlatformToolset="v142"`, or perhaps simply remove it?

**Reply by turbosmurfen** [reply]
> 
> 
> Hi
> 
> I don't think this is a CodeQL issue, but rather an issue with building your code using GH actions. Have you tried adding a simple workflow for just building your code?
> 
> From the error message it appears that passing in `/p:PlatformToolset="v143"` in your build command does not work, because the `windows-latest` machines does not have a matching MSBuild version installed; what happens if you replace it with `/p:PlatformToolset="v142"`, or perhaps simply remove it?

Removing PlatformToolset doesn't help either. GH Actions, is there a site or GitHub for this? I suck at using MSBuild and workflows.

**Reply by hvitved** [reply]
It looks like build tools v143 are included with Visual Studio 2022, so one thing you could try is to replace the `runs-on: windows-latest` with `runs-on: windows-2022` (see https://github.blog/changelog/2021-11-16-github-actions-windows-server-2022-with-visual-studio-2022-is-now-generally-available-on-github-hosted-runners/).

**Reply by turbosmurfen** [reply]
> It looks like build tools v143 are included with Visual Studio 2022, so one thing you could try is to replace the `runs-on: windows-latest` with `runs-on: windows-2022` (see https://github.blog/changelog/2021-11-16-github-actions-windows-server-2022-with-visual-studio-2022-is-now-generally-available-on-github-hosted-runners/).

Thanks for the information. I'm sure this would work. But CodeQL Build Tracer doesn't support Windows 11 or Windows Server 2022. Thanks so much for your help.
